docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -340,6 +342,52 @@ You will also need to configure the `maven-compiler-plugin` to detect the `quark
 </build>
 ----
 
+==== Create new extension modules using Maven
+
+`create-extension` mojo of `io.quarkus:quarkus-maven-plugin` can be used to generate stubs of Maven modules needed for implementing a new Quarkus extension.
+
+WARNING: This mojo can be currently used only for adding extensions to an established source tree, such as https://github.com/quarkusio/quarkus[Quarkus] or https://github.com/quarkusio/quarkus[Camel Quarkus]. Creating extension projects from scratch is not supported yet.
+
+Example: Let's add a new extension called `my-ext` to Quarkus source tree:
+
+[source, shell, subs=attributes+]
+----
+git clone https://github.com/quarkusio/quarkus.git
+cd quarkus
+cd extensions
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create-extension -N -Dquarkus.artifactIdBase=my-ext
```

Comment:
I'm surprised we don't pass the name of the extension as a parameter. This will be useful for the artifact name.

Reply:
The mojo has the `nameBase` option for that https://github.com/quarkusio/quarkus/blob/f957cc8d9089b6e892687b95d812f85dd1bf831c/devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java#L134-L142

If `nameBase` is not set, it is derived from `artifactIdBase`. For `new-ext` the derived nameBase is `New Ext`.

Let me check if/how I should add that info to the guide.

Reply:
I added a paragraph about that in b4067d0

Reply:
I think you should include it in this very example.

Reply:
OK, will do.